include/psa/crypto_values.h

```diff
@@ -1732,8 +1743,45 @@ static inline int mbedtls_svc_key_id_equal( mbedtls_svc_key_id_t id1,
             mbedtls_key_owner_id_equal( id1.owner, id2.owner ) );
 }
 
+/** Check whether a key identifier is null.
+ *
+ * \param key Key identifier.
+ *
+ * \return Non-zero if the key identifier is null, zero otherwise.
+ */
+static inline int mbedtls_svc_key_id_is_null( mbedtls_svc_key_id_t key )
+{
+    return( ( key.key_id == 0 ) && ( key.owner == 0 ) );
+}
+
 #endif /* !MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER */
 
+#define PSA_KEY_HANDLE_INIT MBEDTLS_SVC_KEY_ID_INIT
```

Comment:
> I don't see the point of PSA_KEY_HANDLE_INIT and psa_key_handle_is_null. Using 0 for invalid/unset is a widespread convention and doesn't need named constants.
It is because a handle can be of integer or structure type thus 0 does not work. On that note, if before this openless work, the handles had been initialized with a PSA_KEY_HANDLE_INIT like macro it would have eased this openless work (as I wouldn't have had to introduce it). Thus, "using 0 for invalid/unset is a widespread convention and doesn't need named constants", is maybe not a very good idea if you are not completely sure that the integer value will always be an integer value.

>If they're defined at all, they should not be in crypto_values.h and they should hav an MBEDTLS_ prefix, since they aren't part of the standard API. They should be defined in library/something.h since they are neither part of the standard API, nor a useful API extension (they're new, but key handles are a legacy feature thta is only preserved for backward compatibility), nor a preserved legacy compatibility feature.

I am going to move them to crypto_compat.h along with the definition of `psa_key_handle_t` as they are tightly coupled to the `psa_key_handle_t` type. Because of this tight coupling, I find it strange to change there name. Thus, not doing it yet, but can do if you still prefer after this move to crypto_compat.h.

Reply:
psa_key_handle_equal has been removed.

Reply:
Addressed in commit: "Move handle related definition to crypto_compat.h"